src/providers/windshaft.js

```diff
@@ -1,13 +1,76 @@
 var d3 = require("d3");
+var _ = require("underscore");
+var Providers = require("./");
 
 function WindshaftProvider(options) {
-	this.sql_api_template = options.sql_api_template || 'http://{user}.cartodb.com';
+	this.tiler_template = options.tiler_template || 'http://{user}.cartodb.com';
 	this.user = options.user;
 	this.table = options.table;
 	this.format = options.format;
+	this.options = options;
 	this.tileCache = {};
+	this.initialize();
 }
 
 WindshaftProvider.prototype = {
+	initialize: function(){
+		var self = this;
+		this.tiler_template = this.tiler_template.replace("{user}", this.user);
+		var mapconfig = this._generateMapconfig(this.table);
+		var url = this.tiler_template + "/api/v1/map?config=" + encodeURIComponent(JSON.stringify(mapconfig));
+		cartodb.d3.net.jsonp(url + "&callback=mapconfig", function(data){
+			self.layergroup = data;
+			self.urlTemplate = self.tiler_template + "/api/v1/map/" + self.layergroup.layergroupid+"/0/{z}/{x}/{y}.geojson"
+		});
+	},
 
-};
\ No newline at end of file
+	getTile: function(tilePoint, callback){
+		if(this.layergroup){
+	    var self = this;
+	    var tileData = this.tileCache[tilePoint.zoom + ":" + tilePoint.x + ":" + tilePoint.y];
+	    if (tileData) {
+	      callback(tilePoint, tileData);
+	    }
+	    else{
+	      var url = this.urlTemplate
+	                .replace("{x}", tilePoint.x)
+	                .replace("{y}", tilePoint.y)
+	                .replace("{z}", tilePoint.zoom);
+	      this.getGeometry(url, function(err, geometry){
+	        if(geometry.type === "Topology"){
+	          self.format = "topojson";
+	          geometry = topojson.feature(geometry, geometry.objects.vectile);
+	        }
+	        callback(tilePoint, geometry);
+	      }.bind(this));
+	    }
+	  }
+  },
+
+  getGeometry: function(url, callback){
+      d3.json(url, callback);
+  },
+
+	_generateMapconfig: function(table) {
```

Comment:
We need it to use the library as a standalone library outside cartodb.js, whilst also avoiding the use of the SQL API. Similar thing to what we do in Torque.